### PR TITLE
TECH Added TORNADO_DEBUG option to activate the tracemalloc debugger …

### PR DESCRIPTION
**Pull request by maric-cp**
…to investigate memory usage and leaks.

### CODE REVIEW

File: heroku2elk/heroku2Logstash.py [hunk 1]
```diff
@@ -216,11 +228,21 @@ def configure_logger():
...
         logger.info("Start H2L in single-processing mode")
         app.listen(8080)
 
-    # instantiate an AMQP connection at start to create the queues (needed when logstash starts)
-    tornado.ioloop.IOLoop.instance().add_future(AMQPConnectionSingleton().get_channel(tornado.ioloop.IOLoop.instance()), lambda x: logger.info("pid:{} AMQP is connected".format(os.getpid())))
-    tornado.ioloop.IOLoop.instance().start()
+    # instantiate an AMQP connection at start to create the queues
+    # (needed when logstash starts)
+    ins = tornado.ioloop.IOLoop.instance()
+    tornado.ioloop.IOLoop.instance().add_future(
```
Comment (GillesRasigade) [inline heroku2elk/heroku2Logstash.py]: Factorize the `tornado.ioloop.IOLoop.instance()` with defined `ins` instance

**Review by GillesRasigade — DISMISSED** [review]
One minor comment. Overall LGTM because most of the changes are syntax (auto ?) update.

This project does not includes code coverage (codecov) not clean documentation. This has to be fixed.

**Comment by maric-cp** [comment]
Well without the new module actually you had only those (by hand) cleanup.

File: heroku2elk/heroku2Logstash.py [hunk 2]
```diff
@@ -216,11 +228,20 @@ def configure_logger():
...
-    # instantiate an AMQP connection at start to create the queues (needed when logstash starts)
-    tornado.ioloop.IOLoop.instance().add_future(AMQPConnectionSingleton().get_channel(tornado.ioloop.IOLoop.instance()), lambda x: logger.info("pid:{} AMQP is connected".format(os.getpid())))
-    tornado.ioloop.IOLoop.instance().start()
+    # instantiate an AMQP connection at start to create the queues
+    # (needed when logstash starts)
+    ins = tornado.ioloop.IOLoop.instance()
+    ins.add_future(AMQPConnectionSingleton().get_channel(ins),
+                   lambda x: logger.info("AMQP is connected"))
+    if MainConfig.tornado_debug:
+        from dump_mem import record_top, start
```
Comment (GillesRasigade) [inline heroku2elk/heroku2Logstash.py:243]: This code is not authorized in Python either

File: heroku2elk/heroku2Logstash.py [hunk 3]
```diff
@@ -216,11 +228,20 @@ def configure_logger():
...
+        start()
+        ins.PeriodicCallback(record_top, 1800*10**3)
```
Comment (GillesRasigade) [inline heroku2elk/heroku2Logstash.py:245]: These lines are not tested

File: heroku2elk/dump_mem.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+from logging import getLogger
```
Comment (GillesRasigade) [inline heroku2elk/dump_mem.py:1]: This file is not tested

**Review by GillesRasigade — CHANGES_REQUESTED** [review]

File: heroku2elk/heroku2Logstash.py (hunk 3 above)
Comment (maric-cp) [reply heroku2elk/heroku2Logstash.py:245]: Agreed

File: heroku2elk/dump_mem.py (hunk 4 above)
Comment (maric-cp) [reply heroku2elk/dump_mem.py:1]: This is from python standard library

File: heroku2elk/heroku2Logstash.py (hunk 2 above)
Comment (maric-cp) [reply heroku2elk/heroku2Logstash.py:243]: Why ?

**Review by GillesRasigade — APPROVED** [review]
Tests have to be added on the memory trace part.

**Pull request merged**